tests/test_dynamic_mock.js

```diff
@@ -6,23 +6,63 @@ const nock = require('..')
 
 require('./cleanup_after_each')()
 
-test('one function returning the body defines a full mock', function(t) {
+test('one function not returning an array causes an error', function(t) {
   nock('http://example.test')
     .get('/abc')
     .reply(function() {
       return 'ABC'
     })
 
-  request.get('http://example.test/abc', function(err, resp, body) {
-    if (err) {
-      throw err
-    }
-    t.equal(resp.statusCode, 200)
-    t.equal(body, 'ABC')
+  request.get('http://example.test/abc', function(err) {
+    t.match(
+      err,
+      Error('A single function provided to .reply MUST return an array')
+    )
+    t.end()
+  })
+})
+
+test('one function returning an empty array causes an error', function(t) {
```

Comment:
`tests/test_dynamic_mock.js` should be merged into `tests/test_reply_function_sync.js and `tests/test_reply_function_async.js`. That could be dealt with in a follow-on, though for now could you add the new tests to one of those two files instead?

Reply:
`tests/test_reply_function_sync.js` and `tests/test_reply_function_async.js` should probably be split up into four files that test the reply-body function and reply-full-response function separately. 
I'll move the new tests out of `tests/test_dynamic_mock.js`, but I was not planning on splitting up the other two in this PR. Just FYI.

Reply:
Sounds great.

I'm all for smaller groupings of tests. In my experience it makes them easier to digest, and to check that all the use cases have been covered.